edxlearndot/learndot.py

```diff
@@ -363,14 +399,132 @@ def set_enrolment_status(self, enrolment_id, status):
             log.error(msg)
             raise LearndotAPIException(msg)
 
+        if not unconditional:
+            try:
+                enrolment_status = EnrolmentStatusLog.objects.get(learndot_enrolment_id=enrolment_id)
+                if enrolment_status.status == EnrolmentStatus.PASSED:
+                    log.info(
+                        "Learndot enrolment was logged as set to %s at %s, so skipping update.",
+                        enrolment_status.status,
+                        enrolment_status.updated_at
+                    )
+                    return
+            except EnrolmentStatusLog.DoesNotExist:
+                pass
+
         response = requests.post(
             self.get_enrolment_management_url_template().format(enrolment_id=enrolment_id),
```

Comment:
nit: could this be changed to `self.get_enrolment_management_url(enrolment_id) instead?